client/shared/dev/defaults.bzl

```diff
@@ -0,0 +1,25 @@
+load("@aspect_rules_ts//ts:defs.bzl", _ts_config = "ts_config", _ts_project = "ts_project")
+
+def ts_project(name, composite = True, declaration = True, declaration_map = True, resolve_json_module = True, source_map = True, **kwargs):
```

Comment:
None of the parameters are being used, and I don't think they should be (until we need them I guess, which is hopefully never) so lets just drop them?

Reply:
This is a convenient way to not have to try to pull these out of `kwargs`. They should in theory be override-able if they get set like when you call the actual ts_project.

Reply:
For example, if we hardcode `composite = False` in the rule, then someone passes `composite` in kwargs, we will have declared it twice and that will be an error.

Reply:
I don't think we want anyone to pass `composite` though. I'd rather be strict until we need it, and preferably we never have to override any of these things.

Reply:
They can pass it in with `kwargs` though, and we'd have an error. I think on principle we should keep `kwargs` so that someone can add or customize any attribute they want.

Reply:
You're not using the args though, you hardcoded the values below.

I don't think we should be enabling customization though, lets keep it as closed as possible so people can change/configure as little as possible. In the future this should be done with gazelle so we want to avoid letting things be customized as much as possible other then essentially just the `deps` and things like `tags`, `visibility` etc. (which are all in the kwargs).

Reply:
Doh that was an accident. Meant to use the variables. I think I disagree about keeping it closed because there are always exceptions and I want to allow them to tinker once the migration is over. I think they should be able to use the `ts_project` macro like they would the original. I'm trying to follow the same pattern that's used in Angular: https://github.com/angular/angular/blob/main/tools/defaults.bzl

Reply:
I'd say the opposite lol

We want BUILDs to be generated with gazelle so ideally they never diverge from the standard we define in the macro. The less configurable it is the easier it is to automate.

Reply:
At the very least I'd vote to use the `kwargs.pop` style to do defaults to avoid this giant line, wdyt about doing...
```
def ts_project(name, **kwargs):
  ts_project(
    name = name,
    tsconfig = kwargs.pop("tsconfig", "//:tsconfig"),
    declarations = kwargs.pop("declarations", True),
    ... etc ...
  )
```
?